yarn-project/end-to-end/src/shared/capture_private_execution_steps.ts

```diff
@@ -28,15 +33,31 @@ async function _createClientIvcProofFiles(directory: string, executionSteps: Pri
   };
 }
 
-export async function capturePrivateExecutionStepsIfEnvSet(label: string, interaction: ContractFunctionInteraction) {
+export async function capturePrivateExecutionStepsIfEnvSet(
+  label: string,
+  interaction: ContractFunctionInteraction | DeployMethod,
+  opts?: Omit<ProfileMethodOptions & DeployOptions, 'profileMode'>,
+  expectedSteps?: number,
+) {
   // Not included in env_var.ts as internal to e2e tests.
   const ivcFolder = process.env.CAPTURE_IVC_FOLDER;
   if (ivcFolder) {
     logger.info(`Capturing client ivc execution steps for ${label}`);
-    const result = await interaction.profile({ profileMode: 'execution-steps' });
+    const result = await interaction.profile({ ...opts, profileMode: 'full' });
+    if (expectedSteps !== undefined && result.executionSteps.length !== expectedSteps) {
+      //throw new Error(`Expected ${expectedSteps} execution steps, got ${result.executionSteps.length}`);
```

Comment:
Artifact from debugging?

Reply:
Error enforced now if the number of steps is unexpected